packages/bierzo-wallet/src/logic/account.ts

```diff
@@ -8,10 +8,7 @@ import { getConnectionFor, isBnsSpec } from "./connection";
  * Returns the address associated with the name, or undefined if not registered.
  * The name must include a namespace ("*iov")
  */
-export async function lookupRecipientAddressByName(
-  username: string,
-  chainId: ChainId,
-): Promise<Address | undefined> {
+export async function lookupRecipientAddressByName(username: string, chainId: ChainId): Promise<Address> {
```

Comment:
Can we use return values instead of exceptions? This allows us to differentiate between expected return values and errors.

A return type could be: `Promise<Address | "name_not_found" | "no_address_for_blockchain">`